bee-node/src/plugins/mqtt/config.rs

```diff
@@ -1,34 +1,124 @@
-// Copyright 2020-2021 IOTA Stiftung
+// Copyright 2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
 use serde::Deserialize;
 
-const DEFAULT_ADDRESS: &str = "tcp://localhost:1883";
+use multiaddr::{Multiaddr, Protocol};
+use std::net::{IpAddr, SocketAddr, ToSocketAddrs};
+
+pub(crate) const DEFAULT_SERVER_BIND_ADDR: &str = "/ip4/0.0.0.0/tcp/1883";
+pub(crate) const DEFAULT_CONSOLE_BIND_ADDR: &str = "/ip4/0.0.0.0/tcp/1884";
 
 #[derive(Default, Deserialize)]
 pub struct MqttConfigBuilder {
-    address: Option<String>,
+    server_bind_addr: Option<Multiaddr>,
+    console_bind_addr: Option<Multiaddr>,
 }
 
 impl MqttConfigBuilder {
+    /// Creates a new config builder.
     pub fn new() -> Self {
         Self::default()
     }
 
+    /// Sets the binding address for the MQTT server.
+    pub fn server_bind_addr(mut self, addr: &str) -> Self {
+        match addr.parse() {
+            Ok(addr) => {
+                self.server_bind_addr.replace(addr);
+            }
+            Err(e) => panic!("Error parsing bind address: {:?}", e),
```

Comment:
Why is it ok to panic instead of propagating the error?

Reply:
good find, will fix

Reply:
we panic also in other config builders what I have seen, can you confirm pls @thibault-martinez ?

Reply:
I think since we have a database to protect from corruption it's not a long-term solution to panic. Instead we should introduce an "Emergency shutdown" system, that can be triggered for various situations, like:
(a) certain sockets do not bind (gossip, autopeering, mqtt, ...)
(b) not enough disk space
(c) not enough RAM
...